starboard/shared/starboard/player/filter/audio_frame_discarder.cc

```diff
@@ -23,13 +23,7 @@ namespace player {
 namespace filter {
 
 void AudioFrameDiscarder::OnInputBuffers(const InputBuffers& input_buffers) {
-  if (input_buffer_infos_.size() >= kMaxNumberOfPendingInputBufferInfos) {
-    // This shouldn't happen as it's DCHECKed at the end of this function. Add
-    // an extra check here to ensure that |input_buffer_infos_| won't grow
-    // without bound, which can lead to OOM in production.
-    return;
-  }
-
+  ScopedLock lock(mutex_);
```

Comment:
nit: Should this lock be moved to line 39? It doesn't seem to be needed for the loop.

Reply:
The mutex is to protect |input_buffer_infos_|, so it's needed for line 31.